src/com/amazon/ion/impl/_Private_LSTWriter.java

```diff
@@ -0,0 +1,284 @@
+                case impList:
+                case sym:
+                    state = null;
+                    break;//we need to intern decsymbols on exiting the entire lst declaration(might not have seen import context yet), so this is a no op.
+                case preSym:
+                case impMID:
+                case impVer:
+                case impName:
+                    throw new UnsupportedOperationException();
+            }
+        }
+    }
+
+    public void setFieldName(String name) {
+        if(state == null) {
+            if (name.equals("imports")) {
+                if(seenImports) throw new UnsupportedOperationException("Two import declarations found.");
+                state = State.preImp;
+            } else if (name.equals("symbols")) {
+                if(decSymbols.size() > 0) throw new UnsupportedOperationException("Two symbol list declarations found.");
+                state = State.preSym;
+            } else {
+                throw new UnsupportedOperationException();
+            }
+        } else if(state == State.impDesc) {
+            if(name.equals("name")) {
+                state = State.impName;
+            } else if(name.equals("version")) {
+                state = State.impVer;
+            } else if(name.equals("max_id")) {
+                state = State.impMID;
+            }
+        } else {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    public void writeString(String value) {
+        if(state == null) throw new UnsupportedOperationException("Open content unsupported via the managed binary writer");
+        switch(state) {
+            case sym:
+                decSymbols.add(value);
+                break;
+            case impName:
+                decImports.getLast().name = value;
+                state = State.impDesc;
+                break;
+            default:
+                throw new UnsupportedOperationException("Open content unsupported via the managed binary writer");
+
+        }
+    }
+
+    public void writeSymbol(String content){
+        if(state == State.preImp && content.equals("$ion_symbol_table")) {
+            state = null;
+        } else {
+            throw new UnsupportedOperationException("Open content unsupported via the managed binary writer");
+        }
+    }
+    public void writeSymbolToken(SymbolToken content) throws IOException {
+        writeSymbol(content.getText());
+    }
+
+    public void writeInt(long value) {
+        if(state == State.impVer) {
+            decImports.getLast().version = (int) value;
+            state = State.impDesc;
+        } else if(state == State.impMID) {
+            decImports.getLast().maxID = (int) value;
+            state = State.impDesc;
+        } else {
+            throw new UnsupportedOperationException("Open content is unsupported via these APIs.");
+        }
+    }
+
+    public void writeInt(BigInteger value){
+        writeInt(value.longValue());
```

Comment:
were inherently limited by the size of the symbol array were declaring to hold any SST or LST

Reply:
Can you expand on this? I'm not sure I understand what you're saying. This API may be used to declare a version or max_id of an import, which doesn't necessarily directly relate to the size of an array in memory.

Even if we can't support integers larger than a `long`, it would be much better to detect overflow and throw early than to potentially end up writing out the wrong value or fail in some other cryptic way.

Reply:
If you have more values than can be held in a long there is no collection in java that can be used to contain that number of values. I can add a size check failing early is definitely preferable to corrupting data silently in an unanticipated way.
I do not believe we were verifying the size prior to this submission.

Reply:
You could have a custom linked list, for example, that could contain enough elements to fill all of the memory on the box. It would need a special SymbolTable implementation, but it could be done. But regardless, a declared import need not actually exist in memory because imports are not required to be present in the catalog.

Anyway, a size check sounds fine.